vendor/assets/javascripts/brainstem/loaders/model-loader.coffee

```diff
@@ -34,18 +34,19 @@ class Brainstem.ModelLoader extends Brainstem.AbstractLoader
     storage = @storageManager.storage(@_getCollectionName())
     model = @loadOptions.model
 
-    storage.add(model, remove: false) if model
+    storage.add(model, remove: false) if model && model.id
 
-    @internalObject = model || storage.get(id) || @storageManager.createNewModel(@loadOptions.name, id: id)
-    @externalObject = @internalObject
+    @internalObject = storage.get(id) || @storageManager.createNewModel(@loadOptions.name, id: id)
+    @externalObject = model || @internalObject
 
   _updateStorageManagerFromResponse: (resp) ->
-    @internalObject.parse(resp)
+    attributes = @internalObject.parse(resp)
+    # @externalObject.set(attributes)
```

Comment:
Ugh, no probably not. This is definitely WIP. We probably don't actually want to use a separate `externalObject` reference for the model loader.